### PR TITLE
Fix content deflating for IIS HTTP responses 

### PR DESCRIPTION
**Pull request by blat**
|    Q          |   A
|-------------- | ------
| Bugfix        | yes
| BC Break      | no

### Description

Some response coming from IIS are using `gzuncompress` instead of `gzinflate`.

Note: This fix was already fixed 10 years ago on ZF1 https://github.com/zendframework/zf1/commit/1e158aa35ff672564a74636c140ebc8309d6ed07 but this this was not ported in ZF2.
See: https://framework.zend.com/issues/browse/ZF-12457.html

To reproduce:

```
use Laminas\Http\Client;
$client = new Client('https://www.vnews.com/With-Federal-Money-Running-Out-Lebanon-Starts-to-Pick-Up-Motel-Tabs-for-Homeless-48667855');
echo $client->send()->getBody();
```

Current result:

```
PHP Warning: gzuncompress(): data error in vendor/laminas/laminas-http/src/Response.php on line 653
```

### CODE REVIEW

File: src/Response.php [hunk 1]
```diff
@@ -648,7 +648,7 @@ protected function decodeDeflate($body)
          */
         $zlibHeader = unpack('n', substr($body, 0, 2));
 
-        if ($zlibHeader[1] % 31 === 0) {
+        if ($zlibHeader[1] % 31 === 0 && ord($body[0]) === 0x78 && in_array(ord($body[1]), [0x01, 0x5e, 0x9c, 0xda])) {
```
Comment (Ocramius) [inline src/Response.php:651]: Needs tests, IMO.

Possible to hit this through `fromString()`? 

https://github.com/laminas/laminas-http/blob/40218e9786a471832b4e52a9fe900e0b130aa641/src/Response.php#L203

Comment (Ocramius) [inline src/Response.php:653]: Needs tests, IMO.

Possible to hit this through `fromString()`? 

https://github.com/laminas/laminas-http/blob/40218e9786a471832b4e52a9fe900e0b130aa641/src/Response.php#L203

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/Response.php (hunk 1 above)
Comment (blat) [reply src/Response.php:653]: Hi @Ocramius!

I have added a new test.
Not sure it's possible to move this in `fromString()`.

**Comment by froschdesign** [comment]
@blat 
You are using the wrong branch because [2.17.0](https://github.com/laminas/laminas-http/releases) is the last release, not 2.16.
See also in the contributing guidelines: [What branch to issue the pull request against? ](https://github.com/laminas/.github/blob/995faeffd32f4fbdac62ed2b3f4888074b65ea91/CONTRIBUTING.md#what-branch-to-issue-the-pull-request-against)

**Comment by blat** [comment]
@froschdesign

PHP7 is not supported by 2.17.x release.
No plan for a 2.16.1?

**Comment by froschdesign** [comment]
> PHP7 is not supported by 2.17.x release.

Why? [PHP 7 is dead](https://www.php.net/supported-versions).

**Comment by blat** [comment]
Yes, I know PHP7 will be EOL at the end of the month. I'm current working on this upgrade ;)
In the meantime, a [2.16.1](https://github.com/laminas/laminas-http/milestone/15) will be appreciated!

File: test/ResponseTest.php [hunk 2]
```diff
@@ -288,6 +288,22 @@ public function testNonStandardDeflateResponseLaminas6040()
...
     }
 
+    /**
+     * Make sure there no confusion with complient "deflate" responses.
+     *
+     * @link https://framework.zend.com/issues/browse/ZF-12457.html
+     */
+    public function testStandardDeflateResponseLaminas12457()
+    {
+        $responseTest = file_get_contents(__DIR__ . '/_files/response_deflate_iis_valid');
```
Comment (Ocramius) [inline test/ResponseTest.php:298]: Can we get a smaller example file here, perhaps? :thinking:

File: src/Response.php (hunk 1 above)
Comment (Ocramius) [inline src/Response.php:653]: These functions need to be imported

**Review by Ocramius — CHANGES_REQUESTED** [review]
Some minor adjustments needed.

@froschdesign I'd be OK with merging this to an old-stable, as an exception: the contributor has put time to fix in it, and the merge-up should be straightforward too. WDYT?

**Comment by froschdesign** [comment]
@Ocramius 
> …the contributor has put time to fix in it, and the merge-up should be straightforward too.

I agree with that. 👍🏻

File: src/Response.php (hunk 1 above)
Comment (Ocramius) [reply src/Response.php:653]: Happy with the test as-is, just hoped for a smaller compressed file :D

**Comment by blat** [comment]
I've updated with the smaller example I was able to found.

Thank you!

**Review by Ocramius — APPROVED** [review]
Thanks @blat! Releasing now :)

**Pull request merged**